pex/cli/commands/lock.py

```diff
@@ -1028,3 +1272,235 @@ def process_req_edits(
+            )
+            if isinstance(lock_update, Result):
+                return lock_update
+
+            try_(self._process_lock_update(lock_update, lock_file, lock_file_path))
+        else:
+            targets = try_(
+                self._resolve_targets(
+                    action="creating",
+                    style=self.options.style,
+                    target_configuration=target_configuration,
+                )
+            )
+            lockfile = try_(
+                create(
+                    lock_configuration=lock_configuration,
+                    requirement_configuration=requirement_configuration,
+                    targets=targets,
+                    pip_configuration=pip_configuration,
+                )
+            )
+            with safe_open(lock_file_path, "w") as fp:
+                self._dump_lockfile(lockfile, output=fp)
+
+        sync_target = None  # type: Optional[SyncTarget]
+        if self.options.venv:
+            if os.path.exists(self.options.venv):
+                try:
+                    venv = Virtualenv(self.options.venv)
+                except InvalidVirtualenvError as e:
+                    return Error("The given --venv is not a valid venv: {err}".format(err=e))
+            else:
+                if self.options.venv_python:
+                    interpreter = (
+                        PythonInterpreter.from_binary(self.options.venv_python)
+                        if os.path.isfile(self.options.venv_python)
+                        else PythonInterpreter.from_env(self.options.venv_python)
+                    )
+                else:
+                    targets = target_options.configure(self.options).resolve_targets()
+                    interpreters = [
+                        target.get_interpreter()
+                        for target in targets.unique_targets()
+                        if isinstance(target, LocalInterpreter)
+                    ]
+                    if len(interpreters) != 1:
+                        return Error(
+                            "In order to create the venv {venv} the sync operation needs to know "
+                            "which interpreter to use to create it.\n"
+                            "Use `--venv-interpreter` to select an interpreter for venv creation."
+                        )
+                    interpreter = interpreters[0]
+                venv = Virtualenv.create(self.options.venv, interpreter=interpreter)
+                if self.options.pip:
+                    venv.install_pip(upgrade=True)
+            sync_target = SyncTarget.resolve_command(venv=venv, command=self.passthrough_args)
+        elif self.passthrough_args:
+            sync_target = try_(
+                SyncTarget.resolve_venv(self.passthrough_args[0], *self.passthrough_args[1:])
+            )
+        if not sync_target:
+            return Ok()
+
+        if self.options.dry_run:
```

Comment:
From this, it looks like `--dry-run` only applies to the sync to the venv, and the `--lock` lockfile is still created/mutated, which I find a bit surprising: I'd expect a `--dry-run` flag to mean not mutate anything "real" on disk, only caches and temporary files. It looks like the docs for `--dry-run` say "Don't update the lock file".

Am I missing something?

Reply:
You are, partly. The `--dry-run` option is accessed by helpers that perform the update. The create path, though, does not handle dry-run and always creates the lock file. I'm not sure what to do about that. A dry-run venv mutation makes sense (the create path can sync an existing venv instead of creating a new one). A dry run new lock file creation makes a lot less sense.

Reply:
Ok, there is now a trimmed down dry run report for lock creation such that no lock file hits disk.